### DISCUSSION TITLE
Skia - Setter of DependencyProperty of user control is never called when is bind to a property of the view model.

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#9876] [Q&A]
I have a user control with a IsON dependency property that is bind to the IsON property o the view model. 

ViewModel:

```
[ObservableProperty]
private bool _isOn = false;
```

UserControl:

```
public bool IsON
{
    get { return (bool)GetValue(IsONProperty); }
    set
    {
        SetValue(IsONProperty, value);
        if (value)
        {
            this.ST_border.BorderBrush = BorderColorON;
            this.ST_border.Background = BackgroundColorON;
            this.ST_text.Foreground = TextColorON;
            this.ST_text.Text = ControlTextON;
        }
        else
        {
            this.ST_border.BorderBrush = BorderColorOFF;
            this.ST_border.Background = BackgroundColorOFF;
            this.ST_text.Foreground = TextColorOFF;
            this.ST_text.Text = ControlTextOFF;
        }
    }
}

public static readonly DependencyProperty IsONProperty =
    DependencyProperty.Register(nameof(IsON), typeof(bool), typeof(ST_LED), new PropertyMetadata(false));
```

MainPage:

```
<local:ST_LED x:Name="led"
                Width="100"
                Height="100"
                ControlTextON="POWER ON"
                ControlTextOFF="POWER OFF"
                BorderColorON="#63bf54"
                BackgroundColorON="#3c7333"
                BorderColorOFF="#2e5c8c"
                BackgroundColorOFF="#152a40"
                TextColorON="White"
                TextColorOFF="SlateGray"
                IsON="{x:Bind ViewModel.IsON, Mode=OneWay}" />
```

When ViewModel.IsON changes, the setter of ST_LED.IsON is never called.

It can be checked by setting a breakpoint in the setter.

I have added a minimum example to check it.

NOTE: In windows works fine, only fails on the skia project.
[TestSkiaProperties.zip](https://github.com/unoplatform/uno/files/9592020/TestSkiaProperties.zip)

**Comment by jeromelaban** [comment]
Thanks for the repro. What you're seeing is a slight behavior difference between Uno and WinUI's implementation for x:Bind, where Uno invokes the underlying `DependencyObject.SetValue()` for performance reasons, while WinUI invokes the C# property directly.

While this is an actual difference (that we may end up fixing at some point, https://github.com/unoplatform/uno/issues/9886), you should not be relying on the fact that the setter is invoked. For instance, if you're styling or animating the properties of your control, the Style or Animation engines will use `DependencyObject.SetValue()`, therefore skipping the code in your C# setter.

Best is to use the property changed handlers:

```csharp
public bool IsON
{
    get => (bool)GetValue(IsONProperty);
    set => SetValue(IsONProperty, value);
}

public static readonly DependencyProperty IsONProperty =
    DependencyProperty.Register(
        nameof(IsON), 
        typeof(bool),
        typeof(ST_LED), 
        new PropertyMetadata(
            false, 
            propertyChangedCallback: (s, e) => (s as ST_LED)?.OnPropertyChanged((bool)e.NewValue)));

private void OnPropertyChanged(bool value)
{
    if (value)
    {
        this.ST_border.BorderBrush = BorderColorON;
        this.ST_border.Background = BackgroundColorON;
        this.ST_text.Foreground = TextColorON;
        this.ST_text.Text = ControlTextON;
    }
    else
    {
        this.ST_border.BorderBrush = BorderColorOFF;
        this.ST_border.Background = BackgroundColorOFF;
        this.ST_text.Foreground = TextColorOFF;
        this.ST_text.Text = ControlTextOFF;
    }
}
```

**Reply by mcNets** [reply]
Thanks for the help @jeromelaban.

I'm sorry to bother you with this kind of issues.

**Reply by jeromelaban** [reply]
No need to be sorry! We need to know about these issues.

This is the kind of issues that are not necessarily easy to find because we're used to "common" ways for defining classes and controls, so people will know what to do when they find this.
